2022-11-01 08:33:25 info: ---------------------------------------------
2022-11-01 08:33:25 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-11-01 08:33:25 info: ---------------------------------------------
2022-11-01 08:33:38 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-programmerFun.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js (6.338 s)
  ● programmerFun › HTML should be syntactically valid

    expect(received).toBe(expected) // Object.is equality

    Expected: ""
    Received: "
    inline
      11:2  error  <img> is missing required \"alt\" attribute  wcag/h37·
    ✖ 1 problem (1 error, 0 warnings)
    "

      44 |   const { results } = htmlValidate.validateString(htmlText);
      45 |   const validationReport = stylish(results);
    > 46 |   expect(validationReport).toBe('');
         |                            ^
      47 | }
      48 |
      49 | module.exports = {

      at toBe (test-runner/jsdom-helpers.js:46:28)
      at Object.validateHTML (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:45:5)

  ● programmerFun › should use try/catch

    expect(received).toBeDefined()

    Received: undefined

      57 |
      58 |   test('should use try/catch', () => {
    > 59 |     expect(state.tryCatch).toBeDefined();
         |                            ^
      60 |   });
      61 | });
      62 |

      at Object.toBeDefined (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:59:28)


  ● Test suite failed to run

    TypeError: Cannot read properties of undefined (reading 'json')

      43 |   const response = await requestData('https://xkcd.now.sh/?comic=latest');
      44 |   const data = await response
    > 45 |     .json()
         |      ^
      46 |
      47 |     .then((data) => {
      48 |       renderImage(data);

      at main (file:/C:/Users/lenovo/Desktop/MyHYF/class40/Homework-class40/3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js:45:6)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 3 passed, 5 total
Snapshots:   0 total
Time:        6.518 s, estimated 17 s
Ran all test suites matching /ex1-programmerFun.test.js/i.
2022-11-01 08:33:43 error: *** ESLint Report ***

ex1-programmerFun/index.js
  36:3  warning  The homework tests do not support `innerText`. Please replace with `textContent`  no-restricted-syntax
  44:9  error    'data' is assigned a value but never used                                         no-unused-vars

✖ 2 problems (1 error, 1 warning)


2022-11-01 08:34:44 info: ---------------------------------------------
2022-11-01 08:34:44 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-11-01 08:34:44 info: ---------------------------------------------
2022-11-01 08:34:55 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-programmerFun.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js (5.609 s)
  ● programmerFun › should use try/catch

    expect(received).toBeDefined()

    Received: undefined

      57 |
      58 |   test('should use try/catch', () => {
    > 59 |     expect(state.tryCatch).toBeDefined();
         |                            ^
      60 |   });
      61 | });
      62 |

      at Object.toBeDefined (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:59:28)


  ● Test suite failed to run

    TypeError: Cannot read properties of undefined (reading 'json')

      44 |   const response = await requestData('https://xkcd.now.sh/?comic=latest');
      45 |   const data = await response
    > 46 |     .json()
         |      ^
      47 |
      48 |     .then((data) => {
      49 |       renderImage(data);

      at main (file:/C:/Users/lenovo/Desktop/MyHYF/class40/Homework-class40/3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js:46:6)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 4 passed, 5 total
Snapshots:   0 total
Time:        5.886 s, estimated 7 s
Ran all test suites matching /ex1-programmerFun.test.js/i.
2022-11-01 08:34:59 error: *** ESLint Report ***

ex1-programmerFun/index.js
  37:3  warning  The homework tests do not support `innerText`. Please replace with `textContent`  no-restricted-syntax
  45:9  error    'data' is assigned a value but never used                                         no-unused-vars

✖ 2 problems (1 error, 1 warning)


2022-11-01 08:36:56 info: ---------------------------------------------
2022-11-01 08:36:56 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-11-01 08:36:56 info: ---------------------------------------------
2022-11-01 08:37:06 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-programmerFun.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js (5.272 s)
  programmerFun
    √ HTML should be syntactically valid (239 ms)
    √ should have all TODO comments removed (2 ms)
    × should use `fetch()` (4 ms)
    × should use async/wait (16 ms)
    × should use try/catch (1 ms)

  ● programmerFun › should use `fetch()`

    expect(received).toBeDefined()

    Received: undefined

      48 |
      49 |   test('should use `fetch()`', () => {
    > 50 |     expect(state.fetch).toBeDefined();
         |                         ^
      51 |   });
      52 |
      53 |   test('should use async/wait', () => {

      at Object.toBeDefined (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:50:25)

  ● programmerFun › should use async/wait

    expect(received).toBeDefined()

    Received: undefined

      52 |
      53 |   test('should use async/wait', () => {
    > 54 |     expect(state.async).toBeDefined();
         |                         ^
      55 |     expect(state.await).toBeDefined();
      56 |   });
      57 |

      at Object.toBeDefined (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:54:25)

  ● programmerFun › should use try/catch

    expect(received).toBeDefined()

    Received: undefined

      57 |
      58 |   test('should use try/catch', () => {
    > 59 |     expect(state.tryCatch).toBeDefined();
         |                            ^
      60 |   });
      61 | });
      62 |

      at Object.toBeDefined (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:59:28)

Test Suites: 1 failed, 1 total
Tests:       3 failed, 2 passed, 5 total
Snapshots:   0 total
Time:        5.509 s, estimated 6 s
Ran all test suites matching /ex1-programmerFun.test.js/i.
2022-11-01 08:37:10 error: *** ESLint Report ***

ex1-programmerFun/index.js
  44:3  error  Parsing error: Missing catch or finally clause

✖ 1 problem (1 error, 0 warnings)


2022-11-01 08:40:01 info: ---------------------------------------------
2022-11-01 08:40:01 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-11-01 08:40:01 info: ---------------------------------------------
2022-11-01 08:40:11 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-programmerFun.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js (5.371 s)
  programmerFun
    × HTML should be syntactically valid (230 ms)
    √ should have all TODO comments removed (1 ms)
    √ should use `fetch()` (1 ms)
    √ should use async/wait (1 ms)
    √ should use try/catch (1 ms)

  ● programmerFun › HTML should be syntactically valid

    expect(received).toBe(expected) // Object.is equality

    Expected: ""
    Received: "
    inline
      11:2  error  <h1> cannot be empty, must have text content  empty-heading·
    ✖ 1 problem (1 error, 0 warnings)
    "

      44 |   const { results } = htmlValidate.validateString(htmlText);
      45 |   const validationReport = stylish(results);
    > 46 |   expect(validationReport).toBe('');
         |                            ^
      47 | }
      48 |
      49 | module.exports = {

      at toBe (test-runner/jsdom-helpers.js:46:28)
      at Object.validateHTML (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:45:5)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 4 passed, 5 total
Snapshots:   0 total
Time:        5.628 s, estimated 6 s
Ran all test suites matching /ex1-programmerFun.test.js/i.
2022-11-01 08:40:15 error: *** ESLint Report ***

ex1-programmerFun/index.js
  37:3  warning  The homework tests do not support `innerText`. Please replace with `textContent`  no-restricted-syntax

✖ 1 problem (0 errors, 1 warning)


2022-11-01 08:41:05 info: ---------------------------------------------
2022-11-01 08:41:05 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-11-01 08:41:05 info: ---------------------------------------------
2022-11-01 08:41:15 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-programmerFun.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js (5.393 s)
  programmerFun
    × HTML should be syntactically valid (229 ms)
    √ should have all TODO comments removed (2 ms)
    √ should use `fetch()` (1 ms)
    √ should use async/wait (1 ms)
    √ should use try/catch (1 ms)

  ● programmerFun › HTML should be syntactically valid

    expect(received).toBe(expected) // Object.is equality

    Expected: ""
    Received: "
    inline
      11:2  error  <h1> cannot be empty, must have text content  empty-heading·
    ✖ 1 problem (1 error, 0 warnings)
    "

      44 |   const { results } = htmlValidate.validateString(htmlText);
      45 |   const validationReport = stylish(results);
    > 46 |   expect(validationReport).toBe('');
         |                            ^
      47 | }
      48 |
      49 | module.exports = {

      at toBe (test-runner/jsdom-helpers.js:46:28)
      at Object.validateHTML (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:45:5)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 4 passed, 5 total
Snapshots:   0 total
Time:        5.693 s, estimated 6 s
Ran all test suites matching /ex1-programmerFun.test.js/i.
2022-11-01 08:41:20 error: *** ESLint Report ***

ex1-programmerFun/index.js
  37:3  warning  The homework tests do not support `innerText`. Please replace with `textContent`  no-restricted-syntax

✖ 1 problem (0 errors, 1 warning)


2022-11-01 08:41:45 info: ---------------------------------------------
2022-11-01 08:41:45 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-11-01 08:41:45 info: ---------------------------------------------
2022-11-01 08:41:55 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-programmerFun.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js
  programmerFun
    × HTML should be syntactically valid (235 ms)
    √ should have all TODO comments removed (1 ms)
    √ should use `fetch()` (1 ms)
    √ should use async/wait (1 ms)
    √ should use try/catch (1 ms)

  ● programmerFun › HTML should be syntactically valid

    expect(received).toBe(expected) // Object.is equality

    Expected: ""
    Received: "
    inline
      11:2  error  <h1> cannot be empty, must have text content  empty-heading·
    ✖ 1 problem (1 error, 0 warnings)
    "

      44 |   const { results } = htmlValidate.validateString(htmlText);
      45 |   const validationReport = stylish(results);
    > 46 |   expect(validationReport).toBe('');
         |                            ^
      47 | }
      48 |
      49 | module.exports = {

      at toBe (test-runner/jsdom-helpers.js:46:28)
      at Object.validateHTML (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:45:5)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 4 passed, 5 total
Snapshots:   0 total
Time:        5.224 s, estimated 6 s
Ran all test suites matching /ex1-programmerFun.test.js/i.
2022-11-01 08:41:59 error: *** ESLint Report ***

ex1-programmerFun/index.js
  36:3  warning  The homework tests do not support `innerText`. Please replace with `textContent`  no-restricted-syntax

✖ 1 problem (0 errors, 1 warning)


2022-11-01 08:42:46 info: ---------------------------------------------
2022-11-01 08:42:46 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-11-01 08:42:46 info: ---------------------------------------------
2022-11-01 08:42:57 info: All unit tests passed.
2022-11-01 08:43:07 info: All steps were completed successfully
2022-11-01 09:32:27 info: ---------------------------------------------
2022-11-01 09:32:27 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-11-01 09:32:27 info: ---------------------------------------------
2022-11-01 09:32:56 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-programmerFun.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js (17.028 s)
  programmerFun
    √ HTML should be syntactically valid (876 ms)
    × should have all TODO comments removed (4 ms)
    × should use `fetch()` (3 ms)
    × should use async/wait (8 ms)
    × should use try/catch (1 ms)

  ● programmerFun › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

  ● programmerFun › should use `fetch()`

    expect(received).toBeDefined()

    Received: undefined

      48 |
      49 |   test('should use `fetch()`', () => {
    > 50 |     expect(state.fetch).toBeDefined();
         |                         ^
      51 |   });
      52 |
      53 |   test('should use async/wait', () => {

      at Object.toBeDefined (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:50:25)

  ● programmerFun › should use async/wait

    expect(received).toBeDefined()

    Received: undefined

      52 |
      53 |   test('should use async/wait', () => {
    > 54 |     expect(state.async).toBeDefined();
         |                         ^
      55 |     expect(state.await).toBeDefined();
      56 |   });
      57 |

      at Object.toBeDefined (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:54:25)

  ● programmerFun › should use try/catch

    expect(received).toBeDefined()

    Received: undefined

      57 |
      58 |   test('should use try/catch', () => {
    > 59 |     expect(state.tryCatch).toBeDefined();
         |                            ^
      60 |   });
      61 | });
      62 |

      at Object.toBeDefined (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:59:28)

Test Suites: 1 failed, 1 total
Tests:       4 failed, 1 passed, 5 total
Snapshots:   0 total
Time:        17.681 s
Ran all test suites matching /ex1-programmerFun.test.js/i.
2022-11-01 09:33:05 error: *** ESLint Report ***

ex1-programmerFun/index.js
  56:71  error  Parsing error: Unexpected token it

✖ 1 problem (1 error, 0 warnings)


2022-11-01 09:33:31 info: ---------------------------------------------
2022-11-01 09:33:31 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-11-01 09:33:31 info: ---------------------------------------------
2022-11-01 09:33:45 error: *** Unit Test Error Report ***

Command failed: npx jest ex1-programmerFun.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js (7.988 s)
  programmerFun
    √ HTML should be syntactically valid (717 ms)
    × should have all TODO comments removed (3 ms)
    × should use `fetch()`
    × should use async/wait (3 ms)
    × should use try/catch (1 ms)

  ● programmerFun › should have all TODO comments removed

    expect(received).toBeFalsy()

    Received: true

      103 | function testTodosRemoved(getSource) {
      104 |   test('should have all TODO comments removed', () => {
    > 105 |     expect(/\bTODO\b/.test(getSource())).toBeFalsy();
          |                                          ^
      106 |   });
      107 | }
      108 |

      at Object.toBeFalsy (test-runner/unit-test-helpers.js:105:42)

  ● programmerFun › should use `fetch()`

    expect(received).toBeDefined()

    Received: undefined

      48 |
      49 |   test('should use `fetch()`', () => {
    > 50 |     expect(state.fetch).toBeDefined();
         |                         ^
      51 |   });
      52 |
      53 |   test('should use async/wait', () => {

      at Object.toBeDefined (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:50:25)

  ● programmerFun › should use async/wait

    expect(received).toBeDefined()

    Received: undefined

      52 |
      53 |   test('should use async/wait', () => {
    > 54 |     expect(state.async).toBeDefined();
         |                         ^
      55 |     expect(state.await).toBeDefined();
      56 |   });
      57 |

      at Object.toBeDefined (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:54:25)

  ● programmerFun › should use try/catch

    expect(received).toBeDefined()

    Received: undefined

      57 |
      58 |   test('should use try/catch', () => {
    > 59 |     expect(state.tryCatch).toBeDefined();
         |                            ^
      60 |   });
      61 | });
      62 |

      at Object.toBeDefined (3-UsingAPIs/Week2/unit-tests/ex1-programmerFun.test.js:59:28)

Test Suites: 1 failed, 1 total
Tests:       4 failed, 1 passed, 5 total
Snapshots:   0 total
Time:        8.553 s, estimated 18 s
Ran all test suites matching /ex1-programmerFun.test.js/i.
2022-11-01 09:33:50 error: *** ESLint Report ***

ex1-programmerFun/index.js
  56:71  error  Parsing error: Unexpected token it

✖ 1 problem (1 error, 0 warnings)


2022-11-01 09:38:27 info: ---------------------------------------------
2022-11-01 09:38:27 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-11-01 09:38:27 info: ---------------------------------------------
2022-11-01 09:38:40 info: All unit tests passed.
2022-11-01 09:39:00 info: All steps were completed successfully
2022-11-01 09:39:14 info: ---------------------------------------------
2022-11-01 09:39:14 info: >>> Running Unit Test `ex1-programmerFun` <<<
2022-11-01 09:39:14 info: ---------------------------------------------
2022-11-01 09:39:25 info: All unit tests passed.
2022-11-01 09:39:40 info: All steps were completed successfully
2022-11-01 11:40:06 info: ------------------------------------------
2022-11-01 11:40:06 info: >>> Running Unit Test `ex2-pokemonApp` <<<
2022-11-01 11:40:06 info: ------------------------------------------
2022-11-01 11:40:34 error: *** Unit Test Error Report ***

Command failed: npx jest ex2-pokemonApp.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex2-pokemonApp.test.js (15.49 s)
  pokemonApp
    × HTML should be syntactically valid (623 ms)
    √ should have all TODO comments removed (1 ms)
    √ should use `fetch()` (1 ms)
    √ should use `await fetch()`
    √ should use try/catch

  ● pokemonApp › HTML should be syntactically valid

    expect(received).toBe(expected) // Object.is equality

    Expected: ""
    Received: "
    inline
      12:2  error  <button> is missing required \"type\" attribute  element-required-attributes·
    ✖ 1 problem (1 error, 0 warnings)
    "

      44 |   const { results } = htmlValidate.validateString(htmlText);
      45 |   const validationReport = stylish(results);
    > 46 |   expect(validationReport).toBe('');
         |                            ^
      47 | }
      48 |
      49 | module.exports = {

      at toBe (test-runner/jsdom-helpers.js:46:28)
      at Object.validateHTML (3-UsingAPIs/Week2/unit-tests/ex2-pokemonApp.test.js:40:5)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 4 passed, 5 total
Snapshots:   0 total
Time:        15.935 s, estimated 17 s
Ran all test suites matching /ex2-pokemonApp.test.js/i.
2022-11-01 11:40:57 error: *** Spell Checker Report ***

ex2-pokemonApp/index.js:42:29 - Unknown word (Pokimon)

2022-11-01 11:42:28 info: ------------------------------------------
2022-11-01 11:42:28 info: >>> Running Unit Test `ex2-pokemonApp` <<<
2022-11-01 11:42:28 info: ------------------------------------------
2022-11-01 11:42:49 info: All unit tests passed.
2022-11-01 11:43:01 error: *** ESLint Report ***

ex2-pokemonApp/index.js
  41:3  warning  Remove commented-out code  hyf/no-commented-out-code

✖ 1 problem (0 errors, 1 warning)


2022-11-01 11:43:40 info: ------------------------------------------
2022-11-01 11:43:40 info: >>> Running Unit Test `ex2-pokemonApp` <<<
2022-11-01 11:43:40 info: ------------------------------------------
2022-11-01 11:44:06 info: All unit tests passed.
2022-11-01 11:44:45 info: All steps were completed successfully
2022-11-01 11:45:18 info: ------------------------------------------
2022-11-01 11:45:18 info: >>> Running Unit Test `ex2-pokemonApp` <<<
2022-11-01 11:45:18 info: ------------------------------------------
2022-11-01 11:45:43 info: All unit tests passed.
2022-11-01 11:46:10 info: All steps were completed successfully
2022-11-01 11:48:03 info: ------------------------------------------
2022-11-01 11:48:03 info: >>> Running Unit Test `ex2-pokemonApp` <<<
2022-11-01 11:48:03 info: ------------------------------------------
2022-11-01 11:48:48 info: All unit tests passed.
2022-11-01 11:49:32 info: All steps were completed successfully
2022-11-02 12:31:28 info: -----------------------------------------
2022-11-02 12:31:28 info: >>> Running Unit Test `ex3-rollAnAce` <<<
2022-11-02 12:31:28 info: -----------------------------------------
2022-11-02 12:31:35 info: All unit tests passed.
2022-11-02 12:31:45 info: All steps were completed successfully
2022-11-02 12:48:28 info: ----------------------------------------
2022-11-02 12:48:28 info: >>> Running Unit Test `ex4-diceRace` <<<
2022-11-02 12:48:28 info: ----------------------------------------
2022-11-02 12:48:36 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-diceRace.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js
  ex4-diceRace
    √ should exist and be executable (3 ms)
    √ should have all TODO comments removed (1 ms)
    √ `rollDice` should not contain unneeded console.log calls (1 ms)
    √ should use `dice.map()` (1 ms)
    √ should use `Promise.race()` (1 ms)
    × should resolve as soon as a die settles successfully (4 ms)
    √ should reject with an Error as soon as a die rolls off the table (1 ms)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect(received).toBeInstanceOf(expected)

    Expected constructor: Promise

    Received value has no prototype
    Received value: undefined

      52 |
      53 |     const promise = rollDice();
    > 54 |     expect(promise).toBeInstanceOf(Promise);
         |                     ^
      55 |     const result = await promise;
      56 |     expect(typeof result).toBe('string');
      57 |

      at Object.toBeInstanceOf (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:54:21)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      46 |
      47 |   test('should resolve as soon as a die settles successfully', async () => {
    > 48 |     expect.assertions(3);
         |            ^
      49 |     expect(exported).toBeDefined();
      50 |
      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);

      at Object.assertions (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:48:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        1.677 s, estimated 2 s
Ran all test suites matching /ex4-diceRace.test.js/i.
C:\Users\lenovo\Desktop\MyHYF\class40\Homework-class40\3-UsingAPIs\helpers\pokerDiceRoller.js:64
        reject(new Error(`Die ${die} rolled off the table.\n`));
               ^

Error: Die 1 rolled off the table.

    at rollOnce (C:\Users\lenovo\Desktop\MyHYF\class40\Homework-class40\3-UsingAPIs\helpers\pokerDiceRoller.js:64:16)
    at Timeout._onTimeout (C:\Users\lenovo\Desktop\MyHYF\class40\Homework-class40\3-UsingAPIs\helpers\pokerDiceRoller.js:77:26)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
2022-11-02 12:49:49 info: ----------------------------------------
2022-11-02 12:49:49 info: >>> Running Unit Test `ex4-diceRace` <<<
2022-11-02 12:49:49 info: ----------------------------------------
2022-11-02 12:49:55 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-diceRace.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js
  ex4-diceRace
    √ should exist and be executable (2 ms)
    √ should have all TODO comments removed (1 ms)
    × `rollDice` should not contain unneeded console.log calls (5 ms)
    √ should use `dice.map()` (1 ms)
    √ should use `Promise.race()` (1 ms)
    × should resolve as soon as a die settles successfully (2 ms)
    √ should reject with an Error as soon as a die rolls off the table (2 ms)

  ● ex4-diceRace › `rollDice` should not contain unneeded console.log calls

    expect(received).toBe(expected) // Object.is equality

    Expected: false
    Received: true

      137 |       });
      138 |
    > 139 |     expect(callsConsoleLog).toBe(false);
          |                             ^
      140 |   });
      141 | }
      142 |

      at Object.toBe (test-runner/unit-test-helpers.js:139:29)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect(received).toBeInstanceOf(expected)

    Expected constructor: Promise

    Received value has no prototype
    Received value: undefined

      52 |
      53 |     const promise = rollDice();
    > 54 |     expect(promise).toBeInstanceOf(Promise);
         |                     ^
      55 |     const result = await promise;
      56 |     expect(typeof result).toBe('string');
      57 |

      at Object.toBeInstanceOf (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:54:21)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      46 |
      47 |   test('should resolve as soon as a die settles successfully', async () => {
    > 48 |     expect.assertions(3);
         |            ^
      49 |     expect(exported).toBeDefined();
      50 |
      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);

      at Object.assertions (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:48:12)

Test Suites: 1 failed, 1 total
Tests:       2 failed, 5 passed, 7 total
Snapshots:   0 total
Time:        1.654 s, estimated 2 s
Ran all test suites matching /ex4-diceRace.test.js/i.

  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "NINE".

      17 |   const dice = [1, 2, 3, 4, 5];
      18 |   Promise.race(dice.map((roll) => rollDie(roll)))
    > 19 |     .then((resolve) => console.log(resolve))
         |                                                                         ^
      20 |     .catch((err) => console.log(err));
      21 | }
      22 |

      at console.log (node_modules/@jest/console/build/CustomConsole.js:172:10)
      at 3-UsingAPIs/Week2/homework/ex4-diceRace.js:19:73


  ●  Cannot log after tests are done. Did you forget to wait for something async in your test?
    Attempted to log "Error: Die 1 rolled off the table.

        at rollOnce (C:\Users\lenovo\Desktop\MyHYF\class40\Homework-class40\3-UsingAPIs\helpers\pokerDiceRoller.js:64:16)
        at Timeout._onTimeout (C:\Users\lenovo\Desktop\MyHYF\class40\Homework-class40\3-UsingAPIs\helpers\pokerDiceRoller.js:77:26)
        at listOnTimeout (node:internal/timers:559:17)
        at processTimers (node:internal/timers:502:7)".

      62 |       getRandomNumber(MAX_ROLLS - MIN_ROLLS + 1) + MIN_ROLLS;
      63 |
    > 64 |     logStamped(`Die ${die} scheduled for ${randomRollsToDo} rolls...`);
         |                ^
      65 |
      66 |     let offTable = false;
      67 |

      at rollOnce (3-UsingAPIs/helpers/pokerDiceRoller.js:64:16)
      at Timeout._onTimeout (3-UsingAPIs/helpers/pokerDiceRoller.js:77:26)
      at processTimers (node:internal/timers:502:7)".
      at console.log (node_modules/@jest/console/build/CustomConsole.js:172:10)
      at 3-UsingAPIs/Week2/homework/ex4-diceRace.js:19:108
2022-11-02 12:51:40 info: ----------------------------------------
2022-11-02 12:51:40 info: >>> Running Unit Test `ex4-diceRace` <<<
2022-11-02 12:51:40 info: ----------------------------------------
2022-11-02 12:51:47 error: *** Unit Test Error Report ***

Command failed: npx jest ex4-diceRace.test.js --colors
 FAIL  3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js
  ex4-diceRace
    √ should exist and be executable (3 ms)
    √ should have all TODO comments removed (2 ms)
    √ `rollDice` should not contain unneeded console.log calls (2 ms)
    √ should use `dice.map()` (1 ms)
    √ should use `Promise.race()`
    × should resolve as soon as a die settles successfully (6 ms)
    √ should reject with an Error as soon as a die rolls off the table (2 ms)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect(received).toBeInstanceOf(expected)

    Expected constructor: Promise

    Received value has no prototype
    Received value: undefined

      52 |
      53 |     const promise = rollDice();
    > 54 |     expect(promise).toBeInstanceOf(Promise);
         |                     ^
      55 |     const result = await promise;
      56 |     expect(typeof result).toBe('string');
      57 |

      at Object.toBeInstanceOf (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:54:21)

  ● ex4-diceRace › should resolve as soon as a die settles successfully

    expect.assertions(3)

    Expected three assertions to be called but received two assertion calls.

      46 |
      47 |   test('should resolve as soon as a die settles successfully', async () => {
    > 48 |     expect.assertions(3);
         |            ^
      49 |     expect(exported).toBeDefined();
      50 |
      51 |     const randomSpy = jest.spyOn(Math, 'random').mockReturnValue(0);

      at Object.assertions (3-UsingAPIs/Week2/unit-tests/ex4-diceRace.test.js:48:12)

Test Suites: 1 failed, 1 total
Tests:       1 failed, 6 passed, 7 total
Snapshots:   0 total
Time:        2.137 s
Ran all test suites matching /ex4-diceRace.test.js/i.
C:\Users\lenovo\Desktop\MyHYF\class40\Homework-class40\3-UsingAPIs\helpers\pokerDiceRoller.js:64
        reject(new Error(`Die ${die} rolled off the table.\n`));
               ^

Error: Die 1 rolled off the table.

    at rollOnce (C:\Users\lenovo\Desktop\MyHYF\class40\Homework-class40\3-UsingAPIs\helpers\pokerDiceRoller.js:64:16)
    at Timeout._onTimeout (C:\Users\lenovo\Desktop\MyHYF\class40\Homework-class40\3-UsingAPIs\helpers\pokerDiceRoller.js:77:26)
    at listOnTimeout (node:internal/timers:559:17)
    at processTimers (node:internal/timers:502:7)
2022-11-02 12:52:39 info: ----------------------------------------
2022-11-02 12:52:39 info: >>> Running Unit Test `ex4-diceRace` <<<
2022-11-02 12:52:39 info: ----------------------------------------
2022-11-02 12:52:45 info: All unit tests passed.
2022-11-02 12:52:55 info: All steps were completed successfully
2022-11-02 01:54:15 info: ----------------------------------------
2022-11-02 01:54:15 info: >>> Running Unit Test `ex5-vscDebug` <<<
2022-11-02 01:54:15 info: ----------------------------------------
2022-11-02 01:54:15 warn: A unit test file was not provided.
2022-11-02 01:54:35 info: All steps were completed successfully
